src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -479,6 +477,8 @@ def calculate_sediment_deposition(
                             neighbor_flow_val >> (inflow_offsets[j]*4)) & 0xF
                         if neighbor_flow_weight > 0:
                             f_j = f_raster.get(neighbor_col, neighbor_row)
+                            if f_j == sediment_deposition_nodata:
```

Comment:
We talked about this on a Slack call, as well as this file in general. We're going to try and reach out to Rich to see if he'd be willing to help us comprehend this a bit more.

Reply:
I am pretty sure now that `f_j` will never be nodata when it's accessed here.

At the bottom of this function, `sediment_deposition_raster.set()` and `f_raster.set()` are called. Both these rasters start out filled with nodata, they're only modified in that one place, and they will never be set to nodata. So if `sediment_deposition_raster.get(i)` is defined, then so is `f_raster.get(i)`. 

A pixel is only pushed onto the stack once `sediment_deposition_raster.get(j)` is defined for each of its upstream neighbors `j`. ([code](https://github.com/emlys/invest/blob/1b71cd469ba68e10f103bdf38e416113cf50856c/src/natcap/invest/sdr/sdr_core.pyx#L594-L605))

Therefore I'm pretty sure that `f_j` is also defined for each upstream neighbor. And we only access the upstream neighbors because of the `if neighbor_flow_weight > 0:` check two lines up.

But if there is any uncertainty, it will be safer to leave this nodata check in here.